scripts/parse_live.py

```diff
@@ -72,29 +73,34 @@ def section_name_to_anchor(section):
     section = section.replace('%', '.')
     return section
 
-def query_pageids(wiki, pageids):
+def query_pageids(wiki, pageids, revids):
     params = {
-        'pageids': '|'.join(map(str, pageids)),
         'prop': 'revisions',
-        'rvprop': 'content'
+        'rvprop': 'ids|content'
     }
+    if pageids:
+        params['pageids'] = '|'.join(map(str, pageids))
+    else:
+        params['revids'] = '|'.join(map(str, revids))
+
     for response in self.wiki.query(params):
         for id, page in list(response['query']['pages'].items()):
+            revid = page['revisions'][0]['revid']
```

Comment:
I guess there's always a `'revid'` in the response, even when we don't pass `'revids'` in the request, right?

Reply:
I’m not sure I get your meaning. Do you mean we don't need to pass `revid` argument to `query_article_data`? It's true that there's always a `'revid'` in the response when we pass a `pageid` argument, but for cd_data, we don't have a `pageid`, only a `revid`, so we pass `revid`. Or I misunderstand what you mean? :(

Reply:
Basically I'm worried that this line might throw a `KeyError` if there is no `revisions` or no `revid` keys in the API response.

This function can make either of two kinds of API calls (passing `revids` or `pageids`), so I just wanted to double check that this line is safe to use in either case.

In other words, I'm just double-checking that you tested this CL with both CD and a pageids file :)

Reply:
Oh, I see! Yeah, I've tested with both CD and pageids file. :)